### PR TITLE
fix issue 12950 - Lexer interprets UFCS on hex integer as hex float literal

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: test/runnable/literal.d [hunk 1]
```diff
@@ -225,13 +225,24 @@ void test13907()
...
 
+ulong op12950(ulong v){return v + 12950;}
+
+void test12950()
+{
+    assert(0x00_00_00_01.op12950() == 12951);
+    assert(0x00_00_00_01UL.op12950() == 12951);
+    assert(0b00_00_00_01.op12950() == 12951);
+    assert(0b00_00_00_01UL.op12950() == 12951);
+}
```
Comment (jacob-carlborg) [inline test/runnable/literal.d]: Does this need to be a runnable test? They have more overhead.

**Review by thewilsonator — APPROVED** [review]

File: test/runnable/literal.d (hunk 1 above)
Comment (Basile-z) [reply test/runnable/literal.d]: Yeah but it was the overhead of 1 more test case VS overhead of a few code in an existing one.

Comment (jacob-carlborg) [reply test/runnable/literal.d]: Can't you use `static assert` and put it in `compilable`? Or even better do something like this: https://github.com/dlang/dmd/pull/9963.

**Review by JinShil — APPROVED** [review]
Thanks for fixing these bugs!

File: test/runnable/literal.d (hunk 1 above)
Comment (Basile-z) [reply test/runnable/literal.d]: I think this is nice. The file already exists and from the semantic point of view what's added is very small.

Comment (jacob-carlborg) [reply test/runnable/literal.d]: If the bug fix is in the lexer it doesn't make sense to run the rest of the compiler. Even less compile a executable and run it. Writing a test that only uses the lexer would make the bug fix clear that it only depend on the lexer.

**Comment by Basile-z** [comment]
@JinShil there are two important lists to empty to my eyes:

- [ICE](https://issues.dlang.org/buglist.cgi?keywords=ice&list_id=226415&order=changeddate%20DESC%2Cbug_status%2Cpriority%2Cassigned_to%2Cbug_id&query_format=advanced&resolution=---): only hard stuff remain here. Since January the 40 easiest have been solved. I'm not on this anymore.

- [Reject valid](https://issues.dlang.org/buglist.cgi?bug_status=NEW&component=dmd&keywords=rejects-valid%2C%20&keywords_type=allwords&list_id=225766&product=D&query_format=advanced&resolution=---&version=D2): there are still easy ones in there. *Pick and Fix* !

**Review by WalterBright — APPROVED** [review]

**Comment by Basile-z** [comment]
merge label someone or do you wait 2.086.1 intentionally ?

**Pull request merged**